pkg/asset/agent/installconfig.go

```diff
@@ -150,6 +151,10 @@ func (a *OptionalInstallConfig) validatePlatformsByName(installConfig *types.Ins
 		allErrs = append(allErrs, a.validateBMCConfig(installConfig)...)
 	}
 
+	if installConfig.Platform.Name() == none.Name {
+		allErrs = append(allErrs, a.validateFencingCredentials(installConfig)...)
+	}
```

Comment:
Once you move the validation to pkg/types, you will not need this agent asset code. There is a single code path, where the agent already calls the pkg/types install config validation: 

https://github.com/openshift/installer/blob/74c0c30aa1e7926302b07564480a8eb2f605be20/pkg/asset/agent/installconfig.go#L74